contracts/contracts/MerkleDrop.sol

```diff
@@ -1,20 +1,35 @@
 pragma solidity ^0.5.8;
 
+import "./DroppedToken.sol";
+
 
 contract MerkleDrop {
 
     bytes32 public root;
+    DroppedToken public droppedToken;
+
+    mapping (address => bool) withdrawn;
+
+    event Withdraw(address recipient, uint value);
 
-    constructor(bytes32 _root) public {
+    constructor(DroppedToken _droppedToken, bytes32 _root) public {
         root = _root;
+        droppedToken = _droppedToken;
     }
 
     function withdraw(uint value, bytes32[] memory proof) public {
-        require(checkEntitlement(msg.sender, value, proof), "The proof could not be verified.");
+        withdrawFor(msg.sender, value, proof);
     }
 
     function withdrawFor(address recipient, uint value, bytes32[] memory proof) public {
         require(checkEntitlement(recipient, value, proof), "The proof could not be verified.");
+        require(! withdrawn[recipient], "The recipient has already withdrawn its entitled token.");
+        require(droppedToken.balanceOf(address(this)) > value, "The MerkleDrop does not have tokens to drop yet / anymore.");
```

Comment:
I'd say you should put an `>=` here:
```
droppedToken.balanceOf(address(this)) >= value
```
(or just rely on `droppedToken.transfer` to fail if there are not enough tokens to withdraw, i.e. remove this require statement)